### ISSUE TITLE
Build API form generator service

### ISSUE THREAD

**Issue by dleard** [bcgov/cas-reporting#256]
~Blocked on #270~ Complete

Once the work in #254 is complete & we have the database structures needed to support dynamic form generation from configs, we can build the API endpoint/service that will build these forms dynamically on request.

As a developer
I want an API route/service for form generation
So that when it is called from an emission form it returns a generated schema

As a developer
When I call the API route with the general stationary combustion activity/source types
It returns the correctly generated forms for those inputs

AC:
- [x] The reporting app in django has an api router
- [x] A route & service exists for dynamic form generation (/api/reporting/generate-json-schema?)
- [x] The service pulls a base schema based on the activity-sourceType pair from the base_schema table
- [x] The service dynamically adds the fields from the reporting_fields table related to the gasType/methodology selections
- [x] The service returns a json schema made up of the base schema & dynamic fields
- [x] ~The api route is protected so only logged in users can hit it~ let's do this in #253 to make it easier to start development on that card without protections in place
- [x] pytests for the service

**Comment by dleard** [comment]
Pulling this out of 'in progress' for now in favour of #270 as our understanding of what is required to build the forms has changed. Once #270 is in, some of the work already done here will provide the basis for building the form generator service in a way that is modified to handle the new database models. @pbastia @patriciarussellCAS
